src/pages/transactions/create/Create.tsx

```diff
@@ -85,33 +90,43 @@ export function Create() {
 
     setErrors(undefined);
 
-    setIsSaving(true);
-
-    toast.processing();
-
-    try {
-      await request('POST', endpoint('/api/v1/bank_transactions'), {
-        ...transaction,
-        amount: Number(transaction?.amount),
-        base_type:
-          transaction?.base_type === TransactionType.Deposit
-            ? ApiTransactionType.Credit
-            : ApiTransactionType.Debit,
-      });
-      toast.success(t('created_transaction'));
-      setIsSaving(false);
-      navigate(route('/transactions'));
-    } catch (error) {
-      setIsSaving(false);
-      const axiosError = error as AxiosError;
-      console.error(axiosError);
-
-      if (axiosError?.response?.status === 422) {
-        setErrors(axiosError?.response?.data?.errors);
-        toast.dismiss();
-      } else {
-        toast.error(t('error_title'));
+    if (transaction?.bank_integration_id) {
+      setIsSaving(true);
+
+      toast.processing();
+
+      try {
+        await request('POST', endpoint('/api/v1/bank_transactions'), {
+          ...transaction,
+          amount: Number(transaction?.amount),
+          base_type:
+            transaction?.base_type === TransactionType.Deposit
+              ? ApiTransactionType.Credit
+              : ApiTransactionType.Debit,
+        });
+        toast.success(t('created_transaction'));
+        setIsSaving(false);
+        navigate('/transactions');
+      } catch (error) {
+        setIsSaving(false);
+        const axiosError = error as AxiosError;
+        console.log(axiosError);
+        console.error(axiosError);
+
+        if (axiosError?.response?.status === 422) {
+          setErrors(axiosError?.response?.data?.errors);
+          toast.dismiss();
+        } else {
+          toast.error(t('error_title'));
+        }
+
+        setIsSaving(false);
       }
+    } else {
+      setErrors((prevState) => ({
+        ...prevState,
+        bank_integration_id: t('required_field'),
+      }));
```

Comment:
I don't think we ever do frontend validation. Errors should come from API.

Reply:
API in this case returned me 500 error without message instead of 422 with error message so I have to set it manually.

![image](https://user-images.githubusercontent.com/51542191/204322094-aa8007b1-a968-40c5-87a2-5b193e68538f.png)